docs/build/guides/basics/send-and-receive-payments.mdx

```diff
@@ -7,90 +7,221 @@ description: Learn to send payments and watch for received payments on the Stell
+// is the one that will receive the payment.
+
+const sourceKeys = StellarSdk.Keypair.fromSecret(
   "SCZANGBA5YHTNYVVV4C3U252E2B6P6F5T3U6MM63WBSBZATAQI3EBTQ4",
 );
-var destinationId = "GA2C5RFPE6GCKMY3US5PAB6UZLKIGSPIUKSLRB6Q723BM2OARMDUYEJ5";
-// Transaction will hold a built transaction we can resubmit if the result is unknown.
-var transaction;
+const destinationId =
+  "GA2C5RFPE6GCKMY3US5PAB6UZLKIGSPIUKSLRB6Q723BM2OARMDUYEJ5";
 
 // First, check to make sure that the destination account exists.
-// You could skip this, but if the account does not exist, you will be charged
-// the transaction fee when the transaction fails.
-server
-  .loadAccount(destinationId)
-  // If the account is not found, surface a nicer error message for logging.
-  .catch(function (error) {
+await server.loadAccount(destinationId).catch((error) => {
+  if (error instanceof StellarSdk.NotFoundError) {
+    throw new Error("The destination account does not exist!");
+  } else throw error;
+});
+
+// Now we also load the source account to build the transaction.
+const sourceAccount = await server
+  .loadAccount(sourceKeys.publicKey())
+  .catch((error) => {
     if (error instanceof StellarSdk.NotFoundError) {
       throw new Error("The destination account does not exist!");
-    } else return error;
-  })
-  // If there was no error, load up-to-date information on your account.
-  .then(function () {
-    return server.loadAccount(sourceKeys.publicKey());
-  })
-  .then(function (sourceAccount) {
-    // Start building the transaction.
-    transaction = new StellarSdk.TransactionBuilder(sourceAccount, {
-      fee: StellarSdk.BASE_FEE,
-      networkPassphrase: StellarSdk.Networks.TESTNET,
-    })
-      .addOperation(
-        StellarSdk.Operation.payment({
-          destination: destinationId,
-          // Because Stellar allows transaction in many currencies, you must
-          // specify the asset type. The special "native" asset represents Lumens.
-          asset: StellarSdk.Asset.native(),
-          amount: "10",
-        }),
-      )
-      // A memo allows you to add your own metadata to a transaction. It's
-      // optional and does not affect how Stellar treats the transaction.
-      .addMemo(StellarSdk.Memo.text("Test Transaction"))
-      // Wait a maximum of three minutes for the transaction
-      .setTimeout(180)
-      .build();
-    // Sign the transaction to prove you are actually the person sending it.
-    transaction.sign(sourceKeys);
-    // And finally, send it off to Stellar!
-    return server.submitTransaction(transaction);
-  })
-  .then(function (result) {
-    console.log("Success! Results:", result);
-  })
-  .catch(function (error) {
-    console.error("Something went wrong!", error);
-    // If the result is unknown (no response body, timeout etc.) we simply resubmit
-    // already built transaction:
-    // server.submitTransaction(transaction);
+    } else throw error;
   });
+
+try {
```

Comment:
Putting large pieces of code inside a `try/catch` is bad practice, can we keep them scoped to specific function calls so we can check for the possible error types? Looking at the code block, its not clear to me where errors could be thrown.